src/ngraph/runtime/reference/quantize.hpp

```diff
@@ -44,11 +47,64 @@ namespace ngraph
                 {
                     Coordinate scale_offset_coord = project(input_coord, axes);
 
-                    // apply scale and offset
-                    REAL qvalue =
-                        std::round(input[input_transform.index(input_coord)] /
-                                   scale[scale_offset_transform.index(scale_offset_coord)]) +
-                        offset[scale_offset_transform.index(scale_offset_coord)];
+                    // apply scale
+                    REAL qvalue = input[input_transform.index(input_coord)] /
+                                  scale[scale_offset_transform.index(scale_offset_coord)];
+
+                    auto fe_round_mode = std::fegetround();
+
+                    // round
+                    if (round_mode == op::Quantize::RoundMode::HALF_AWAY_FROM_ZERO)
```

Comment:
You should make this a switch and add gcc flags inline to make sure you cover all cases.